### PR TITLE
plugin/ruby `shellcheck`

### PR DESCRIPTION
**Pull request by gaelicWizard**
## Description
Quote some things and handle unbound parameters.

## Motivation and Context
This was part of my `_command_exists` branch (#1938) but was out of scope.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code follows the code style of this project.
- [x] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] If I have added a new file, I also added it to ``clean_files.txt`` and formatted it using ``lint_clean_files.sh``.
- [ ] I have added tests to cover my changes, and all the new and existing tests pass.

### CODE REVIEW

File: plugins/available/ruby.plugin.bash
Comment (davidpfarrell) [inline plugins/available/ruby.plugin.bash:16]: re `"${1?}"`

* Maybe check the value _before_ invoking `gem list`
```
: "${1?}"
gem list | grep "${1}" | awk '{ print $1; }' | xargs sudo gem uninstall
```

* Maybe also check for empty values vs just unset:
```
: "${1:?}"
```

* Maybe add a more useful error message:
```
: "${1:?no gem name provided}"
```

Even with this, the generated error message would be:
```
bash: 1: no gem name provided
```
Which feels a _tad_ cryptic but does get the message across.

**Review by davidpfarrell — CHANGES_REQUESTED** [review]
Greetings !

The fixes to the ruby plugin seems mostly legit.  Always glad to get more files into the `clean_files.txt` !

I have one comment on the `"${1?}"` usage.

Also, what are the empty files in this PR?  Permission changes?  They don't seem releveant to this PR.

If this is a stacked PR waiting on another PR to merge first, then lets label it DRAFT for now ...

File: plugins/available/ruby.plugin.bash
Comment (gaelicWizard) [reply plugins/available/ruby.plugin.bash:16]: The `${1?}` usage validates the parameter *before* invoking any commands, so gem list won't be executed if it fails.

There's currently no handling of this parameter being empty at all, but what I often do is something like `"${1?${FUNCNAME[0]}: no gem name provided}"` which would result in:
```
bash: 1: remove_gem: no gem name provided
```

**Comment by gaelicWizard** [comment]
> Also, what are the empty files in this PR? Permission changes? They don't seem releveant to this PR.
> 
> If this is a stacked PR waiting on another PR to merge first, then lets label it DRAFT for now ...

I always mark it a draft in the GitHub UI and add "DRAFT" to the title until it's parent branch is approved. I've just rebased this on master and it's now ready to go.

File: plugins/available/ruby.plugin.bash
Comment (gaelicWizard) [reply plugins/available/ruby.plugin.bash]: How's this look now?

**Review by davidpfarrell — APPROVED** [review]
Best I can tell this looks baked and ready to go !

**Review by NoahGorny — APPROVED** [review]
Lgtm
Great work @gaelicWizard and great review @davidpfarrell :smile:

**Pull request merged**
